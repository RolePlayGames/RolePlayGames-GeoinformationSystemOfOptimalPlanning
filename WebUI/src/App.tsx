import { SideMenu } from './menu/SideMenu';
import { Box, styled } from '@mui/material';
import { Route, Routes } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import { CUSTOMERS, FILM_RECIPES, FILM_TYPES, ORDERS, PRODUCTION_LINES } from './routes/routes';
import { CustomersPage } from './customers/CustomersPage';
import { FilmTypesPage } from './film-types/FilmTypesPage';
import { FilmRecipesPage } from './film-recipes/FilmRecipesPage';
import { OrdersPage } from './orders/OrdersPage';
import { ProductionLinesPage } from './production-lines/ProductionLinesPage';

const AppContainer = styled(Box)({
	backgroundColor: '#F3F7FA',
	background: '#11101d',
	minHeight: '100vh',
	display: 'flex',
});

const ToastHost = () => (
	<ToastContainer
		position="top-right"
		autoClose={5000}
		hideProgressBar={false}
		newestOnTop={false}
		closeOnClick
		rtl={false}
		pauseOnFocusLoss
		draggable
		pauseOnHover
		theme="colored"/>
);

const WorkspaceContainer = styled(Box)({
	flexGrow: 1,
	p: 3,
	background: '#e4e9f7',
});

export const App = () => (
	<AppContainer>
		<ToastHost/>
		<SideMenu/>
		<WorkspaceContainer>
			<Routes>
				<Route path={CUSTOMERS} element={<CustomersPage/>}/>
				<Route path={`${CUSTOMERS}/:id`} element={<CustomersPage/>}/>
				<Route path={FILM_TYPES} element={<FilmTypesPage/>}/>
				<Route path={`${FILM_TYPES}/:id`} element={<FilmTypesPage/>}/>
				<Route path={FILM_RECIPES} element={<FilmRecipesPage/>}/>
				<Route path={`${FILM_RECIPES}/:id`} element={<FilmRecipesPage/>}/>
				<Route path={ORDERS} element={<OrdersPage/>}/>
				<Route path={`${ORDERS}/:id`} element={<OrdersPage/>}/>
				<Route path={PRODUCTION_LINES} element={<ProductionLinesPage/>}/>
				<Route path={`${PRODUCTION_LINES}/:id`} element={<ProductionLinesPage/>}/>
			</Routes>
		</WorkspaceContainer>
	</AppContainer>
)
