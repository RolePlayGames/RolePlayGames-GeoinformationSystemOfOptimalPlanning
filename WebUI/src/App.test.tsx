import { App } from './App';
import renderer from 'react-test-renderer';
import { CssBaseline } from '@mui/material';
import { BrowserRouter } from 'react-router-dom';
import { Fragment } from 'react/jsx-runtime';

test('App renders correctly', () => {
	// arrange & act
	const tree = renderer.create(
		<Fragment>
			<CssBaseline/>
			<BrowserRouter>
				<App/>
			</BrowserRouter>
		</Fragment>
	).toJSON();
  
	// assert
	expect(tree).toMatchSnapshot();
});