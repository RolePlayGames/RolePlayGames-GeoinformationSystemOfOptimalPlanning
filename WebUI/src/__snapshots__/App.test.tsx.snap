// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="MuiBox-root css-189l79s"
>
  <div
    className="Toastify"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <div
    className="MuiDrawer-root MuiDrawer-docked css-1he5ohj-MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
    >
      <div
        className="MuiBox-root css-qe3103"
      >
        <div
          className="MuiBox-root css-1xdhyk6"
        >
          <div
            className="MuiBox-root css-j1tw4c"
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-q5mgmq-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                />
              </svg>
            </button>
          </div>
          <ul
            className="MuiList-root MuiList-padding css-1cgtesq-MuiList-root"
          >
            <li
              className="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
              disabled={false}
            >
              <div
                className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1hcncrt-MuiButtonBase-root-MuiListItemButton-root"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  className="MuiListItemIcon-root css-1f8djn6-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14j970f-MuiSvgIcon-root"
                    data-testid="PeopleAltOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.67 13.13C18.04 14.06 19 15.32 19 17v3h4v-3c0-2.18-3.57-3.47-6.33-3.87M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4c-.47 0-.91.1-1.33.24C14.5 5.27 15 6.58 15 8s-.5 2.73-1.33 3.76c.42.14.86.24 1.33.24m-6 0c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4m0-6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m0 7c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4m6 5H3v-.99C3.2 16.29 6.3 15 9 15s5.8 1.29 6 2z"
                    />
                  </svg>
                </div>
                <div
                  className="MuiListItemText-root css-1cag644-MuiListItemText-root"
                >
                  <span
                    className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Customers
                  </span>
                </div>
              </div>
            </li>
          </ul>
        </div>
        <div
          className="MuiBox-root css-tqr95d"
        >
          <div
            className="MuiBox-root css-1mcn6si"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-jbr2j4-MuiSvgIcon-root"
              data-testid="AccountBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-7 3c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m7 13H5v-.23c0-.62.28-1.2.76-1.58C7.47 15.82 9.64 15 12 15s4.53.82 6.24 2.19c.48.38.76.97.76 1.58z"
              />
            </svg>
            <div
              className="MuiBox-root css-1q1j2q5"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 css-pxvdsv-MuiTypography-root"
              >
                User Name
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1 css-14b1zzu-MuiTypography-root"
              >
                User role
              </p>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-kj8sms-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LogoutOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17 8-1.41 1.41L17.17 11H9v2h8.17l-1.58 1.58L17 16l4-4zM5 5h7V3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h7v-2H5z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-1mny8xg"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-z2eky3-MuiTypography-root"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Rhoncus dolor purus non enim praesent elementum facilisis leo vel. Risus at ultrices mi tempus imperdiet. Semper risus in hendrerit gravida rutrum quisque non tellus. Convallis convallis tellus id interdum velit laoreet id donec ultrices. Odio morbi quis commodo odio aenean sed adipiscing. Amet nisl suscipit adipiscing bibendum est ultricies integer quis. Cursus euismod quis viverra nibh cras. Metus vulputate eu scelerisque felis imperdiet proin fermentum leo. Mauris commodo quis imperdiet massa tincidunt. Cras tincidunt lobortis feugiat vivamus at augue. At augue eget arcu dictum varius duis at consectetur lorem. Velit sed ullamcorper morbi tincidunt. Lorem donec massa sapien faucibus et molestie ac.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-z2eky3-MuiTypography-root"
    >
      Consequat mauris nunc congue nisi vitae suscipit. Fringilla est ullamcorper eget nulla facilisi etiam dignissim diam. Pulvinar elementum integer enim neque volutpat ac tincidunt. Ornare suspendisse sed nisi lacus sed viverra tellus. Purus sit amet volutpat consequat mauris. Elementum eu facilisis sed odio morbi. Euismod lacinia at quis risus sed vulputate odio. Morbi tincidunt ornare massa eget egestas purus viverra accumsan in. In hendrerit gravida rutrum quisque non tellus orci ac. Pellentesque nec nam aliquam sem et tortor. Habitant morbi tristique senectus et. Adipiscing elit duis tristique sollicitudin nibh sit. Ornare aenean euismod elementum nisi quis eleifend. Commodo viverra maecenas accumsan lacus vel facilisis. Nulla posuere sollicitudin aliquam ultrices sagittis orci a.
    </p>
  </div>
</div>
`;
